environment/src/test/java/com/sequenceiq/environment/environment/service/EnvironmentModificationServiceTest.java

```diff
@@ -380,6 +383,71 @@ void editByNameParametersExistedAndNotValid() {
         verify(parametersService, never()).saveParameters(environment, parameters);
     }
 
+    @Test
+    void testEditByNameGcpEncryptionResourcesThrowsErrorWhenKeyValidationFails() {
+        ParametersDto parameters = ParametersDto.builder()
+                .withAccountId(ACCOUNT_ID)
+                .withGcpParameters(GcpParametersDto.builder()
+                        .withEncryptionParameters(GcpResourceEncryptionParametersDto.builder()
+                                .withEncryptionKey("dummyEncryptionKey")
+                                .build())
+                        .build())
+                .build();
+        EnvironmentEditDto environmentDto = EnvironmentEditDto.builder()
+                .withAccountId(ACCOUNT_ID)
+                .withParameters(parameters)
+                .build();
+        Environment environment = new Environment();
+        environment.setAccountId(ACCOUNT_ID);
+        BaseParameters baseParameters = new GcpParameters();
+        baseParameters.setId(123L);
+        ValidationResult validationResultError = ValidationResult.builder().error("Wrong key format").build();
+        when(environmentService.getValidatorService()).thenReturn(validatorService);
+        when(validatorService.validateEncryptionKey("dummyEncryptionKey", ACCOUNT_ID)).thenReturn(validationResultError);
+        when(validationResult.hasError()).thenReturn(true);
```

Comment:
I do not think this is needed. The above line depends on `validationResultError` and it will return `true` for `hasError()`.